Task/src/app/components/user-details/user-details.component.ts

```diff
@@ -10,11 +12,18 @@ import {UsersService} from "../../services/users.service";
 })
 export class UserDetailsComponent implements OnInit {
 
-  loggedInUser: User;
+  loggedInUserId: number | undefined;
+  user: User | undefined;
+  private routeSub: Subscription;
 
-  constructor(private usersService : UsersService) {}
+  constructor(private usersService : UsersService,
+              private authService: AuthService,
+              private route: ActivatedRoute) {}
 
   ngOnInit(){
-    this.loggedInUser = this.usersService.getLoggedInUser();
+    this.routeSub = this.route.params.subscribe((params) => {
+      this.user = this.usersService.getUserById(params['id']);
+      this.loggedInUserId = this.authService.getLoggedInUserId()
+    })
```

Comment:
nelabai tikslinga šioje vietoje subscribinti prie url`id, nes jam pasikeitus, paprasčiausiai per naują persikrautų puslapis ir per naują ngOnInit būtų vykdomas. Pakanka naudoti `params = this.route.snapshot.paramMap.get('id');`